src/test/s3/commands/uploadFile.test.ts

```diff
@@ -58,31 +197,182 @@ describe('uploadFileCommand', function () {
+        selection.label = fileLocation.fsPath
+        const response = await getFileToUpload(fileLocation, window, prompt)
 
-        assert.ok(window.message.error?.includes('Failed to upload file'))
+        assert.strictEqual(response, fileLocation)
+    })
+
+    it("prompts quick pick with open file option, and browse for more files, prompts user for other file if that option is selected", async function (){
+        selection.label = "Browse for more files..."
+        
+        const response = await getFileToUpload(fileLocation, window, prompt)
+        assert.strictEqual(response, fileLocation)
+
+    })
+
+    it("returns undefined on second prompt if no file is selected", async function () {
+        selection.label = "Browse for more files..." 
+        window = new FakeWindow({ dialog: { openSelections: undefined } })
+        
+        const response = await getFileToUpload(fileLocation, window, prompt)
+        assert.strictEqual(response, undefined)
+    })
+    
+})
+
+describe('promptUserForBucket',async function () {
+    const fileLocation = vscode.Uri.file('/file.jpg')    
+
+    let s3: S3Client
+    let buckets: S3.Bucket[]
+    let window: FakeWindow
+
+    const promptUndef: <T extends vscode.QuickPickItem>(opts: {
+        picker: vscode.QuickPick<T>
+        onDidTriggerButton?(
+            button: vscode.QuickInputButton,
+            resolve: (value: T[] | PromiseLike<T[] | undefined> | undefined) => void,
+            reject: (reason?: any) => void
+        ): void
+    }) => Promise<T[] | undefined> = () => {
+        return new Promise((resolve, reject) => {
+            resolve(undefined)
+        })
+    }
+    const selection: any = { 
+        label: 'bucket selected',
+        bucket: { Name: 'bucket 1' }
+    }
+    const promptSelec: <T extends vscode.QuickPickItem>(opts: {
+        picker: vscode.QuickPick<T>
+        onDidTriggerButton?(
+            button: vscode.QuickInputButton,
+            resolve: (value: T[] | PromiseLike<T[] | undefined> | undefined) => void,
+            reject: (reason?: any) => void
+        ): void
+    }) => Promise<T[] | undefined> = () => {
+        return new Promise((resolve, reject) => {
+            resolve([selection])
+        })
+    }
+
+    beforeEach(function () {
+        s3 = mock()
+        buckets = [{Name: 'bucket 1'}, {Name: 'bucket 2'}, {Name: 'bucket 3'}]
+        window = new FakeWindow({ dialog: { openSelections: [fileLocation] } })
+    })
 
-        assert.strictEqual(commands.command, 'aws.refreshAwsExplorerNode')
-        assert.deepStrictEqual(commands.args, [node])
+    
+    it("Returns selected bucket", async function () {
+        when(s3.listAllBuckets()).thenResolve(buckets)
+
+        const response = await promptUserForBucket(instance(s3), window, promptSelec)
+        assert.deepStrictEqual(response, buckets[0])
+
+    })
+
+    it("Returns back when selected", async function()  {
```

Comment:
`'returns "back" when selected'` -- ensure that future devs knows that "back" is a string that is returned. Same below with `'return "cancel"...'`